flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/maintenance/stream/MaintenanceTaskBuilder.java

```diff
@@ -0,0 +1,238 @@
+   */
+  public T scheduleOnPosDeleteRecordCount(long posDeleteRecordCount) {
+    triggerEvaluator.posDeleteRecordCount(posDeleteRecordCount);
+    return (T) this;
+  }
+
+  /**
+   * After a given number of new equality delete files since the last run, starts the downstream
+   * job.
+   *
+   * @param eqDeleteFileCount after the downstream job should be started
+   * @return for chained calls
+   */
+  public T scheduleOnEqDeleteFileCount(int eqDeleteFileCount) {
+    triggerEvaluator.eqDeleteFileCount(eqDeleteFileCount);
+    return (T) this;
+  }
+
+  /**
+   * After a given number of new equality delete records since the last run, starts the downstream
+   * job.
+   *
+   * @param eqDeleteRecordCount after the downstream job should be started
+   * @return for chained calls
+   */
+  public T scheduleOnEqDeleteRecordCount(long eqDeleteRecordCount) {
+    triggerEvaluator.eqDeleteRecordCount(eqDeleteRecordCount);
+    return (T) this;
+  }
+
+  /**
+   * After a given time since the last run, starts the downstream job.
+   *
+   * @param interval after the downstream job should be started
+   * @return for chained calls
+   */
+  public T scheduleOnInterval(Duration interval) {
+    triggerEvaluator.timeout(interval);
+    return (T) this;
+  }
+
+  /**
+   * The suffix used for the generated {@link org.apache.flink.api.dag.Transformation}'s uid.
+   *
+   * @param newUidSuffix for the transformations
+   * @return for chained calls
+   */
+  public T uidSuffix(String newUidSuffix) {
+    this.uidSuffix = newUidSuffix;
+    return (T) this;
+  }
+
+  /**
+   * The {@link SingleOutputStreamOperator#slotSharingGroup(String)} for all the operators of the
+   * generated stream. Could be used to separate the resources used by this task.
+   *
+   * @param newSlotSharingGroup to be used for the operators
+   * @return for chained calls
+   */
+  public T slotSharingGroup(String newSlotSharingGroup) {
+    this.slotSharingGroup = newSlotSharingGroup;
+    return (T) this;
+  }
+
+  /**
+   * Sets the parallelism for the stream.
+   *
+   * @param newParallelism the required parallelism
+   * @return for chained calls
+   */
+  public T parallelism(int newParallelism) {
+    this.parallelism = newParallelism;
+    return (T) this;
+  }
+
+  @Internal
+  int id() {
+    return index;
+  }
+
+  @Internal
+  String name() {
+    return name;
+  }
+
+  @Internal
+  TableLoader tableLoader() {
+    return tableLoader;
+  }
+
+  @Internal
+  String uidSuffix() {
+    return uidSuffix;
+  }
+
+  @Internal
+  String slotSharingGroup() {
+    return slotSharingGroup;
+  }
+
+  @Internal
+  Integer parallelism() {
+    return parallelism;
+  }
+
+  @Internal
+  TriggerEvaluator evaluator() {
+    return triggerEvaluator.build();
+  }
+
+  @Internal
+  DataStream<TaskResult> append(
+      DataStream<Trigger> sourceStream,
+      int maintenanceTaskIndex,
+      String maintainanceTaskName,
+      TableLoader newTableLoader,
+      String mainUidSuffix,
```

Comment:
what does `main` mean here?

Reply:
There is a possibility to inherit the suffix and the slotSharingGroup from the TableMaintenance.Builder. It could be overwritten on task-by-task basis.

The `main` here is the value inherited from the `TableMaintenance.Builder`

Reply:
is `default` more accurate than `main`?

Anyway, this is a minor issue since `append` is non-public.

Reply:
Changed to `default`